src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -50,7 +51,8 @@ private MyParameterSet EffectiveParameterSet
                     {
                         _effectiveParameterSet = MyParameterSet.RandomListItem;
                     }
-                    else if (ParameterSetName.Equals(GetRandomCommand.RandomListItemParameterSet, StringComparison.OrdinalIgnoreCase))
+                    else if (ParameterSetName.Equals(GetRandomCommand.RandomListItemParameterSet, StringComparison.OrdinalIgnoreCase) ||
+                             ParameterSetName.Equals(GetRandomCommand.ShuffleSet, StringComparison.OrdinalIgnoreCase))
```

Comment:
Where a condition like this is necessary, most other (recent) areas of the codebase have it with the `||` starting the following line. Also, you can just do a straight `==` on these. Not sure why the original code bothered to check without case sensitivity; the parameter set names shouldn't end up in a different casing to how they're defined as far as I'm aware. Something like the following should be sufficient:

```csharp
else if (ParameterSetName == GetRandomCommand.RandomListItemParameterSet
    || ParameterSetName == GetRandomCommand.ShuffleSet)
```

Reply:
Honestly just in general I have no idea why this code is written in this way, I can't think of a case where this is especially important but no reason to go changing the rest of this now I suppose. Never really thought checking parameter set names would be performance-intensive. 🤔